server/unittools.cpp

```diff
@@ -2586,53 +2578,41 @@ void kill_unit(struct unit *pkiller, struct unit *punit, bool vet)
                         ftc_server,
                         // TRANS: "Cannon ... the Polish Destroyer."
-                        _("%s lost to an attack by the %s %s."), punit_link,
-                        nation_adjective_for_player(pvictor), pkiller_link);
+                        _("%s lost when the %s %s defeated your %s."),
+                        punit_link, nation_adjective_for_player(pvictor),
+                        pkiller_link, punit_link);
         } else {
           fc_assert(other_killed[i] != punit);
           notify_player(player_by_number(i), ptile, E_UNIT_LOST_DEF,
                         ftc_server,
                         /* TRANS: "Cannon lost when the Polish Destroyer
                          * attacked the German Musketeers." */
-                        _("%s lost when the %s %s attacked the %s %s."),
+                        _("%s lost when the %s %s defeated the %s %s."),
                         unit_link(other_killed[i]),
                         nation_adjective_for_player(pvictor), pkiller_link,
                         nation_adjective_for_player(pvictim), punit_link);
         }
       } else if (num_killed[i] > 1) {
         if (i == player_index(pvictim)) {
           int others = num_killed[i] - 1;
-
-          if (others == 1) {
-            notify_player(
-                player_by_number(i), ptile, E_UNIT_LOST_DEF, ftc_server,
-                /* TRANS: "Musketeers (and Cannon) lost to an
-                 * attack from the Polish Destroyer." */
-                _("%s (and %s) lost to an attack from the %s %s."),
-                punit_link, unit_link(other_killed[i]),
-                nation_adjective_for_player(pvictor), pkiller_link);
-          } else {
-            notify_player(
-                player_by_number(i), ptile, E_UNIT_LOST_DEF, ftc_server,
-                /* TRANS: "Musketeers and 3 other units lost to
-                 * an attack from the Polish Destroyer."
-                 * (only happens with at least 2 other units) */
-                PL_("%s and %d other unit lost to an attack "
-                    "from the %s %s.",
-                    "%s and %d other units lost to an attack "
-                    "from the %s %s.",
-                    others),
-                punit_link, others, nation_adjective_for_player(pvictor),
-                pkiller_link);
-          }
+          notify_player(
+              player_by_number(i), ptile, E_UNIT_LOST_DEF, ftc_server,
+              /* TRANS: "Musketeers and 3 other units lost to
+               * an attack from the Polish Destroyer."
+               * (only happens with at least 2 other units) */
+              PL_("%d other unit lost when the %s %s defeated your %s.",
+                  "%d other units lost when the %s %s defeated your %s.",
+                  others),
+              others, nation_adjective_for_player(pvictor), pkiller_link,
+              punit_link);
```

Comment:
Originally, you wanted to have a different message when only one unit is killed. The `%d` can be removed from the first message if it is the *last* argument of `notify_player`. The `%n$x` syntax can be used to reorder arguments when building the string:
```suggestion
              PL_("One other unit lost when the %s %s defeated your %s.",
                  "%4$d other units lost when the %s %s defeated your %s.",
                  others),
              punit_link, nation_adjective_for_player(pvictor), pkiller_link,
              others);
```
Now `punit_link` is number 1, `nation_adjective_for_player(pvictor)` is number 2, `pkiller_link` is number 3, and `others` is number 4.

When formatting the first string, arguments 1 to 3 will be used and 4 will be ignored. When formatting the second strings, all 4 arguments will be used.